src/components/MobileWalletDialog/routes/RenameWallet.tsx

```diff
@@ -0,0 +1,98 @@
+import { Button, FormControl, FormErrorMessage, Input } from '@chakra-ui/react'
+import { useCallback, useState } from 'react'
+import { useTranslate } from 'react-polyglot'
+import { useHistory, useLocation } from 'react-router'
+import { DialogBackButton } from 'components/Modal/components/DialogBackButton'
+import { DialogBody } from 'components/Modal/components/DialogBody'
+import { DialogFooter } from 'components/Modal/components/DialogFooter'
+import {
+  DialogHeader,
+  DialogHeaderLeft,
+  DialogHeaderMiddle,
+} from 'components/Modal/components/DialogHeader'
+import { DialogTitle } from 'components/Modal/components/DialogTitle'
+import { SlideTransition } from 'components/SlideTransition'
+import { Text } from 'components/Text'
+import { updateWallet } from 'context/WalletProvider/MobileWallet/mobileMessageHandlers'
+import type { MobileLocationState } from 'context/WalletProvider/MobileWallet/types'
+
+import { MobileWalletDialogRoutes } from '../types'
+
+const isValidLabel = (label: unknown): label is string => {
+  return typeof label === 'string' && label.length > 0 && label.length < 65
+}
+
+export const RenameWallet = () => {
+  const location = useLocation<MobileLocationState>()
+  const history = useHistory()
+  const translate = useTranslate()
+  const [isSubmitting, setIsSubmitting] = useState(false)
+  const [label, setLabel] = useState<string | null>(null)
+  const [error, setError] = useState<string | null>(null)
+
+  const handleClick = useCallback(async () => {
+    setIsSubmitting(true)
+    try {
+      if (
+        isValidLabel(label) &&
+        location.state.vault?.id &&
+        // Ask the mobile app to update the label on the wallet
+        (await updateWallet(location.state.vault.id, { label }))
```

Comment:
This shouldn't live in an if statement, the two conditions above are valid checks, that one isn't a check - should be part of the body on l.42